build/lib/compilation.ts

```diff
@@ -80,7 +82,10 @@ function createCompile(src: string, build: boolean, emitError: boolean, transpil
 				includeContent: !!build,
 				sourceRoot: overrideOptions.sourceRoot
 			}))
-			.pipe(tsFilter.restore)
+			.pipe(build
+				? tsFilter.restore
+				: tsFilter.restore.pipe(jsFilter).pipe(util.appendOwnPathSourceURL()).pipe(jsFilter.restore)
+			)
```

Comment:
I feel like the following would be easier to read/maintain:

```suggestion
			.pipe(tsFilter.restore)
			.pipe(jsFilter)
			.pipe(build ? es.through() : util.appendOwnPathSourceURL())
			.pipe(jsFilter.restore)
```

Reply:
The difference is that `util.appendOwnPathSourceURL` should only run on the files that were originally filtered out. This suggestion would have it run on the files that the compilation emits as well, which is not desirable and would cause issues since it'd mess up the source map paths.

Could do some extra filtering in the `jsFilter` to have it exclude `out/`, though I think I'd find the existing control flow easier to follow...

Reply:
I moved things. I also made a small helper `util.$if` to allow more compact expressions of conditional steps than `filter()/.restore`. Let me know what you think.

Reply:
I like it!